terra-workspace-manager/sa.tf

```
@@ -1,16 +1,30 @@
-resource "google_service_account" "app" {
+resource "google_service_account" "sqlproxy" {
   count = var.enable ? 1 : 0
 
   provider     = google.target
   project      = var.google_project
-  account_id   = "${local.service}-${local.owner}"
-  display_name = "${local.service}-${local.owner}"
+  account_id   = "${local.service}-${local.owner}-sqlproxy"
+  display_name = "${local.service}-${local.owner}-sqlproxy"
 }
 resource "google_project_iam_member" "app_roles" {
-  count = var.enable ? length(local.sa_roles) : 0
-
   provider = google.target
```

You still want to only create this resource when the module is enabled
```suggestion
  count = var.enable ? 1 : 0
  
  provider = google.target
```